pkg/volume/util/quota/extfs/quota_extfs.go

```diff
@@ -0,0 +1,153 @@
+// +build linux
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extfs
+
+/*
+#include <stdlib.h>
+#include <dirent.h>
+#include <linux/fs.h>
+#include <linux/quota.h>
+#include <errno.h>
+
+#ifndef PRJQUOTA
+#define PRJQUOTA	2
+#endif
+#ifndef Q_SETPQUOTA
+#define Q_SETPQUOTA (unsigned) QCMD(Q_SETQUOTA, PRJQUOTA)
+#endif
+#ifndef Q_GETPQUOTA
+#define Q_GETPQUOTA (unsigned) QCMD(Q_GETQUOTA, PRJQUOTA)
+#endif
+*/
+import "C"
+
+import (
+	"fmt"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/volume/util/quota/common"
+)
+
+//  ext4fs empirically has a maximum quota size of 2^48 - 1 1KiB blocks (256 petabytes)
+const (
+	linuxExtfsMagic       = 0xef53
+	quotaBsize            = 1024                                   // extfs specific
+	bitsPerWord           = 32 << (^uint(0) >> 63)                 // either 32 or 64
+	maxQuota        int64 = (1<<(bitsPerWord-1) - 1) & (1<<58 - 1) // either 1<<31 - 1 or 1<<58 - 1
+)
+
+// VolumeProvider supplies a quota applier to the generic code.
+type VolumeProvider struct {
```

Comment:
I don't think you need the struct here or the interface in quota_linux_common.go above.  Just use a public constructor function: `NewQuotaApplier(...`, and define it in each build.  As an example, see the `NewCgroupManager(...` function in both [cgroup_manager_linux.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cgroup_manager_linux.go#L194) and [cgroup_manager_unsupported.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cgroup_manager_unsupported.go#L33)

Reply:
OK.